dot/telemetry/telemetry.go

```
@@ -57,126 +54,81 @@ var (
-func (h *Handler) SendNetworkData(data *NetworkData) {
-	h.Lock()
-	defer h.Unlock()
-	payload := log.Fields{"bandwidth_download": data.rateIn, "bandwidth_upload": data.rateOut, "msg": "system.interval", "peers": data.peers}
-	telemetryLogger := log.WithFields(log.Fields{"id": 1, "payload": payload, "ts": time.Now()})
-	telemetryLogger.Print()
+// SendMessage sends Message to connected telemetry listeners
+func (t *Handler) SendMessage(msg *Message) {
+	t.msg <- *msg
 }
 
-// BlockIntervalData struct to hold data for block system.interval message
-type BlockIntervalData struct {
-	BestHash           common.Hash
-	BestHeight         *big.Int
-	FinalizedHash      common.Hash
-	FinalizedHeight    *big.Int
-	TXCount            int
-	UsedStateCacheSize int
-}
-
-// SendBlockIntervalData send block data system interval information to telemetry connection
-func (h *Handler) SendBlockIntervalData(data *BlockIntervalData) {
-	h.Lock()
-	defer h.Unlock()
-	payload := log.Fields{"best": data.BestHash.String(), "finalized_hash": data.FinalizedHash.String(), // nolint
-		"finalized_height": data.FinalizedHeight, "height": data.BestHeight, "msg": "system.interval", "txcount": data.TXCount, // nolint
-		"used_state_cache_size": data.UsedStateCacheSize}
-	telemetryLogger := log.WithFields(log.Fields{"id": 1, "payload": payload, "ts": time.Now()})
-	telemetryLogger.Print()
-}
-
-func (h *Handler) sender() {
+func (t *Handler) startListening() {
 	for {
-		h.RLock()
-		line, err := h.buf.ReadBytes(byte(10)) // byte 10 is newline character, used as delimiter
-		h.RUnlock()
-		if err != nil {
-			continue
-		}
-
-		for _, c := range h.wsConn {
-			err := c.WriteMessage(websocket.TextMessage, line)
+		msg := <-t.msg
+		for _, v := range t.connections {
+			err := v.wsconn.WriteMessage(websocket.TextMessage, msgToBytes(msg))
 			if err != nil {
 				// TODO (ed) determine how to handle this error
 				fmt.Printf("ERROR connecting to telemetry %v\n", err)
 			}
```

can you ignore this error for now - I sometimes see a lot of logs of this

---

Removed printf.